packages/graphql-codegen-cli/src/utils/patterns.ts

```diff
@@ -119,7 +119,7 @@ export const makeGlobalPatternSet = (initialContext: CodegenContext) => {
   return {
     watch: sortPatterns([
       ...(typeof config.watch === 'boolean' ? [] : normalizeInstanceOrArray<string>(config.watch ?? [])),
-      relative(process.cwd(), initialContext.filepath),
+      ...(initialContext.filepath ? [relative(process.cwd(), initialContext.filepath)] : []),
```

Comment:
Admittedly, this is not the cleanest way to conditionally add the path to the array. I'm following the above precedence, but I'm happy to refactor it so that the conditions checked when building the array are easier to read.